src/components/RocketPage.js

```diff
@@ -7,8 +7,8 @@ const RocketPage = () => {
   const { isLoading, rocketItems } = useSelector((store) => store.rockets);
   const dispatch = useDispatch();
   useEffect(() => {
-    dispatch(getRocketItems());
-  }, [dispatch]);
+    if (rocketItems.length === 0) { dispatch(getRocketItems()); }
+  });
```

Comment:
I see that you're fetching data after every render,  kindly consider using a dependency array as a second parameter for the useEffect hook, This means that after every render cycle, every effect defined in the corresponding component is executed one after the other based on the positioning in the source code.  [check this link to read more about useEffect without using dependency arra ](https://blog.logrocket.com/useeffect-hook-complete-guide/#:~:text=Again%2C%20if%20you%20do%20not,positioning%20in%20the%20source%20code.)